wayshot/src/wayshot.rs

```diff
@@ -96,67 +66,121 @@ fn main() -> Result<(), Box<dyn Error>> {
+                let x2: i32 = cmp::min(
+                    output.dimensions.x + output.dimensions.width,
+                    region.x_coordinate + region.width,
+                );
+                let y2: i32 = cmp::min(
+                    output.dimensions.y + output.dimensions.height,
+                    region.y_coordinate + region.height,
+                );
 
-            let width = x2 - x1;
-            let height = y2 - y1;
+                let width = x2 - x1;
+                let height = y2 - y1;
 
-            if !(width <= 0 || height <= 0) {
-                intersecting_outputs.push(output);
+                if !(width <= 0 || height <= 0) {
+                    let true_x = region.x_coordinate - output.dimensions.x;
+                    let true_y = region.y_coordinate - output.dimensions.y;
+                    let true_region = CaptureRegion {
+                        x_coordinate: true_x,
+                        y_coordinate: true_y,
+                        width: region.width,
+                        height: region.height,
+                    };
+                    intersecting_outputs.push(IntersectingOutput {
+                        output: output.wl_output.clone(),
+                        region: true_region,
+                    });
+                }
             }
+            if intersecting_outputs.is_empty() {
+                log::error!("Provided capture region doesn't intersect with any outputs!");
+                exit(1);
+            }
+
+            for ouput_info in intersecting_outputs {
+                framecopys.push(libwayshot::capture_output_frame(
+                    &mut globals,
+                    &mut conn,
+                    cursor_overlay,
+                    ouput_info.output.clone(),
+                    Some(ouput_info.region),
+                )?);
+            }
+            (framecopys, Some((region.width, region.height)))
         }
-        if intersecting_outputs.is_empty() {
-            log::error!("Provided capture region doesn't intersect with any outputs!");
-            exit(1);
-        }
-        // NOTE: Figure out box bounds for multi monitor screenshot.
-
-        libwayshot::capture_output_frame(
-            &mut globals,
-            &mut conn,
-            cursor_overlay,
-            output,
-            Some(region),
-        )?
-    } else {
-        libwayshot::capture_output_frame(&mut globals, &mut conn, cursor_overlay, output, None)?
+        CaptureInfo::Output(output) => (
+            vec![libwayshot::capture_output_frame(
+                &mut globals,
+                &mut conn,
+                cursor_overlay,
+                output,
+                None,
+            )?],
+            None,
+        ),
     };
-
     let extension = if args.is_present("extension") {
         let ext: &str = &args.value_of("extension").unwrap().trim().to_lowercase();
+        log::debug!("Using custom extension: {:#?}", ext);
```

Comment:
if case is stdout, won't the log be mixed with buffer in stdout which is the part of image?

Reply:
They'd need to turn on debug flag using `-d`. If you're mixing that with `--stdout` it is bound to disturb the image data.

Reply:
Actually this is a really good point. We should probably make it such that `--stdout` and `--debug` are in conflict with each other. What do you think? @Decodetalkers

Reply:
I think it is necessary, from my opinion..